### PR TITLE
feat(module-tools): add new plugin-module-doc package

### PR DESCRIPTION
**Pull request by 10Derozan**
## Description
模块工程提供文档功能

快速体验
```
cd tests/integration/module/doc
pnpm dev
```

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/plugin-module-doc/index.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:root {
```
Comment (chenjiahan) [inline packages/cli/plugin-module-doc/index.css:1]: 这个文件为啥要放 src 外面

File: packages/cli/plugin-module-doc/package.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    "@scripts/jest-config": "workspace:*",
+    "@types/doctrine": "^0.0.5",
+    "@types/jest": "^27",
+    "@types/node": "^14",
+    "jest": "^27",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "typescript": "^4"
+  },
+  "peerDependencies": {
```
Comment (chenjiahan) [inline packages/cli/plugin-module-doc/package.json:62]: @arco-design/web-react 是不是也要加到 deps or peer deps 里

File: packages/cli/plugin-module-doc/package.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "@modern-js/plugin-module-doc",
+  "description": "A Progressive React Framework for modern web development.",
```
[the remark this answers is not in the document]
Comment (10Derozan) [reply packages/cli/plugin-module-doc/package.json:3]: ok

File: packages/cli/plugin-module-doc/index.css (hunk 1 above)
Comment (10Derozan) [reply packages/cli/plugin-module-doc/index.css:1]: 一般文档项目的全局样式是放外面的，对于这个插件还是放里面吧

**Pull request merged**